Updated preprocess_text.py to fix NeMo-text-processing :: WARNING :: Your input is too long and could take a long time to normalize. Use split_text_into_sentences() to make the input shorter and then call normalize_list().

Adesoji1 opened the pull request:
# What does this PR do ?


Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.


## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

Adesoji1 commented:
@cuichenx

Adesoji1 commented:
> > @cuichenx
> 
> I don't work on the TTS collection. @titu1994 could you reassign?

Sure i will, @XuesongYang  kindly view my PR

akoumpa commented on scripts/dataset_processing/tts/preprocess_text.py:
```diff
@@ -102,14 +116,44 @@ def _process_entry(
-        if lower_case_norm:
-            text = text.lower()
-        text = normalizer.normalize(text, punct_pre_process=True, punct_post_process=True)
+        # Define additional split symbols to enhance splitting
```
hi can you put this whole if statement's body in a separate function and also add CLI flag --enable-split or similar to only run it if the user requested?

Adesoji1 replied:
@akoumpa i have updated the PR with this request, kindly view

akoumpa commented on scripts/dataset_processing/tts/preprocess_text.py:
```diff
@@ -102,14 +116,44 @@ def _process_entry(
 
     if lower_case:
-        text = text.lower()
+        normalized_text = normalized_text.lower()
```
I think you can do without introducing the normalized_text variable and just write at the `text` variable in line 149.

Adesoji1 replied:
@akoumpa  i have updated the PR with this request, kindly view

Adesoji1 replied:
@akoumpa  i have updated the PR with this request, kindly view

Adesoji1 commented:
> This PR is stale because it has been open for 14 days with no activity. Remove stale label or comment or update or this will be closed in 7 days.

@akoumpa pls can you review what I have worked on?

akoumpa commented on scripts/dataset_processing/tts/preprocess_text.py:
```diff
@@ -87,24 +102,66 @@ def get_args():
     normalized_text_key: str,
     lower_case: bool,
     lower_case_norm: bool,
+    enable_split: bool,
```
can you add a default value to False here? I want to avoid breaking any existing code that does not have the `enable_split` parameter.

Adesoji1 replied:
@akoumpa  i have added default = False, kindly view it

Adesoji1 commented:
Sure, I will do that

On Wed, 19 Feb 2025, 6:48 am Alexandros Koumparoulis, <
***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In scripts/dataset_processing/tts/preprocess_text.py
> <https://github.com/NVIDIA/NeMo/pull/11780#discussion_r1960989316>:
>
> >  def _process_entry(
>      entry: dict,
>      normalizer: Normalizer,
>      text_key: str,
>      normalized_text_key: str,
>      lower_case: bool,
>      lower_case_norm: bool,
> +    enable_split: bool,
>
> can you add a default value to False here? I want to avoid breaking any
> existing code that does not have the enable_split parameter.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NVIDIA/NeMo/pull/11780#pullrequestreview-2625689446>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE4COXH64NKW45SQDM7WLL32QQLJPAVCNFSM6AAAAABUYOUBSOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMRVGY4DSNBUGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

akoumpa commented on scripts/dataset_processing/tts/preprocess_text.py:
```diff
@@ -87,24 +102,66 @@ def get_args():
+    # Log sentences exceeding max_num_words words (for debugging)
+    for idx, sentence in enumerate(split_sentences):
+        word_count = len(sentence.split())
+        if word_count > max_num_words:
```
does this ever happen?

Adesoji1 replied:
@akoumpa kindly view my PR

Adesoji1 replied:
@okuchaiev kindly help view my PR too

Adesoji1 commented:
yes it does, try it out.

On Thu, 20 Feb 2025 at 20:50, Alexandros Koumparoulis <
***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In scripts/dataset_processing/tts/preprocess_text.py
> <https://github.com/NVIDIA/NeMo/pull/11780#discussion_r1964244274>:
>
> > +    sentences = normalizer.split_text_into_sentences(text, additional_split_symbols=additional_split_symbols)
> +    split_sentences = []
> +    for sentence in sentences:
> +        words = sentence.split()
> +        if len(words) > max_num_words:
> +            # Split into chunks of max_num_words words
> +            for i in range(0, len(words), max_num_words):
> +                chunk = ' '.join(words[i : i + max_num_words])
> +                split_sentences.append(chunk)
> +        else:
> +            split_sentences.append(sentence)
> +
> +    # Log sentences exceeding max_num_words words (for debugging)
> +    for idx, sentence in enumerate(split_sentences):
> +        word_count = len(sentence.split())
> +        if word_count > max_num_words:
>
> does this ever happen?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NVIDIA/NeMo/pull/11780#pullrequestreview-2630983998>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE4COXC2LT2JVMA5BLH6A3T2QYWYLAVCNFSM6AAAAABUYOUBSOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMZQHE4DGOJZHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
